Sources/NIOPosix/Bootstrap.swift

```diff
@@ -521,6 +521,200 @@ public final class ServerBootstrap {
 // MARK: Async bind methods
 
 extension ServerBootstrap {
+
+    /// Represents a target address or socket for binding a server socket.
+    ///
+    /// `BindTarget` provides a type-safe way to specify different types of binding targets
+    /// for server bootstraps. It supports various address types including network addresses,
+    /// Unix domain sockets, VSOCK addresses, and existing socket handles.
+    public struct BindTarget: Sendable {
+
+        enum Base {
+            case hostAndPort(host: String, port: Int)
+            case address(SocketAddress)
+            case unixDomainSocketPath(String)
+            case vsock(VsockAddress)
+            case socket(NIOBSDSocket.Handle)
+        }
+
+        var base: Base
+        
+        /// Creates a binding target for a hostname and port.
+        ///
+        /// This method creates a target that will resolve the hostname and bind to the
+        /// specified port. The hostname resolution follows standard system behavior
+        /// and may resolve to both IPv4 and IPv6 addresses depending on system configuration.
+        ///
+        /// - Parameters:
+        ///   - host: The hostname or IP address to bind to. Can be a domain name like 
+        ///           "localhost" or "example.com", or an IP address like "127.0.0.1" or "::1"
+        ///   - port: The port number to bind to (0-65535). Use 0 to let the system 
+        ///           choose an available port
+        public static func hostAndPort(_ host: String, _ port: Int) -> BindTarget {
+            BindTarget(base: .hostAndPort(host: host, port: port))
+        }
+
+        /// Creates a binding target for a specific socket address.
+        ///
+        /// Use this method when you have a pre-constructed ``SocketAddress`` that
+        /// specifies the exact binding location, including IPv4, IPv6, or Unix domain addresses.
+        ///
+        /// - Parameter address: The socket address to bind to
+        public static func address(_ address: SocketAddress) -> BindTarget {
```

Comment:
NIT: Since it might be confused with a DNS address
```suggestion
        public static func socketAddress(_ socketAddress: SocketAddress) -> BindTarget {
```